esm/interface/event-target.js

```diff
@@ -1,11 +1,44 @@
 // https://dom.spec.whatwg.org/#interface-eventtarget
 
-import EventTarget from '@ungap/event-target';
+const wm = new WeakMap();
+
+function dispatch({options, target, listener}) {
+  if (options && options.once)
+    target.removeEventListener(this.type, listener);
+  if (typeof listener === 'function') {
+    listener.call(target, this);
+  } else {
+    listener.handleEvent(this);
+  }
+  return this._stopImmediatePropagationFlag;
+}
+
+function invokeListeners({currentTarget, target}) {
+  const secret = wm.get(currentTarget);
+  const listeners = secret && secret[this.type];
+  if (listeners) {
+    if (currentTarget === target) {
+      this.eventPhase = this.AT_TARGET;
+    } else {
+      this.eventPhase = this.BUBBLING_PHASE;
+    }
+    this.currentTarget = currentTarget;
+    this.target = target;
+    listeners.slice(0).some(dispatch, this);
+    delete this.currentTarget;
+    delete this.target;
+  }
+  return this.cancelBubble;
+}
 
 /**
  * @implements globalThis.EventTarget
  */
-class DOMEventTarget extends EventTarget {
+class DOMEventTarget {
+
+  constructor() {
+    wm.set(this, Object.create(null));
```

Comment:
we need a `null` Dictionary class here, possibly in a file a part.

```js
class Dictionary {}
Dictionary.prototype = Object.create(null);
```

Alternatively, I'd like to use `new Map` and related map utilities to handle `secret`. These are performance improvements, so that if none of this is your cup of tea, at least use `const {create} = Object` so no lookup and no patches to `Object.create` can ever affect the result.

Reply:
I find that suggestion interesting and will update the PR 👍

Reply:
I couldn't follow the suggestion exactly as Dictionary.prototype was read only and that code was showing an error,
but I tried something else. There is also a change to use Map, not sure if the 1:1 replacement was what you had in mind and I didn't want to stray too far from the original code.